src/opm/input/eclipse/Schedule/Well/Connection.cpp

```diff
@@ -263,6 +275,7 @@ const std::optional<std::pair<double, double>>& Connection::perf_range() const {
     }
 
     void Connection::scaleWellPi(double wellPi) {
+        this->m_wpimult = wellPi;
```

Comment:
Thanks for pointing this out! The `wpimult` variable is now multiplied in case of multiple `WELPI` and/or `WPIMULT` keywords. I also added a new test in `ScheduleTests.cpp` to check different combinations of `CSKIN`, `WELPI`, `WPIMULT` and `COMPDAT`(which resets `wpimult` to default = 1.0). It seems to work the way I understand these keywords.